content/usgs-packages/sbtools_Get.Rmd

```diff
@@ -0,0 +1,179 @@
+
+Don't forget to load the library if you're in a new R session!
+
+```{r sbtools-library, message=FALSE, warning=FALSE}
+library(sbtools)
+```
+
+```{r sbtools-auth, echo=FALSE}
+# run vizlab::storeSBcreds() once before this can work
+home <- path.expand('~')
+sbCreds <- file.path(home, ".vizlab/sbCreds")
+credList <- readRDS(sbCreds)
+un <- rawToChar(credList$username)
+pw <- rawToChar(credList$password)
+sbtools::authenticate_sb(un, pw)
+```
+
+## Authentication 
+
+This section is specific to authentication with ScienceBase. If you don't have a ScienceBase account, skip to the next section. Just know that you will only be able to download public data.
+
+The first step to authenticating (or logging in) to ScienceBase is to use the function `authenticate_sb`. The arguments are your username and password. Alternatively, you can use the function interactively by not supplying any arguments. It will prompt you for your username in the R console and then your password in a pop-up window. Be very cautious when using the username and password arguments - don't include these in any scripts! To be safe, you can leave out the arguments and use the interactive login. Try interactively logging in:
+
+```{r sbtools-login, eval=FALSE}
+authenticate_sb()
+```
+
+To double check that your authentication was successful, use the function `is_logged_in`. It will return a logical to let you know if you are logged in or not. No arguments are needed.
+
+```{r sbtools-verifylogin}
+is_logged_in()
+```
+
+Each user has a specific ScienceBase id associated with their account. The user ids can be used to inspect what top-level items saved under your account (discussed in next section). To determine your user id, use the function `user_id` in an authenticated session. No arguments are necessary.
+
+```{r sbtools-userid}
+user_id()
+```
+
+When you're done with your session, you can actively logout using the `session_logout`. No arguments are required. If you do not do this, you will be automatically logged out after a certain amount of time or when you close R.
+
+## Inspect and download items
+
+The first inspection step for ScienceBase items is to determine if the item even exists. To do this, use the function `identifier_exists`. The only required argument is `sb_id` which can be either a character string of the item id or an `sbitem`. It will return a logical to indicate if the item exists or not.
+
+```{r sbtools-identifierexists}
+identifier_exists("4f4e4acae4b07f02db67d22b")
+identifier_exists("thisisnotagoodid")
+```
+
+ScienceBase items can be described by alternative identifiers, e.g. digital object identifiers, IPDS codes, etc. They are defined on ScienceBase with a scheme, type, and key. For examples of identifiers, see the "Additional Information | Identifiers" section of [Differential Heating](https://www.sciencebase.gov/catalog/item/580587a2e4b0824b2d1c1f23). 
```

Comment:
noticing that this section is partially redundant with line 203 of PR #271 (https://github.com/USGS-R/training-curriculum/pull/271/files#diff-41534c65bd2e578b204b437073f26e2dR203). could possibly move the formal, full definition of identifiers (with examples) into https://github.com/USGS-R/training-curriculum/blob/master/content/usgs-packages/sbtools_sbitem.Rmd. it'll probably still be good to have a reminder here and in sbtools_Modify.Rmd, but these could become shorter definitions that refer back to sbtools_sbitem.Rmd